cmd/thanos/compact.go

```diff
@@ -81,16 +83,30 @@ func registerCompact(m map[string]setupFunc, app *kingpin.Application) {
 	conf.registerFlag(cmd)
 
 	m[component.Compact.String()] = func(g *run.Group, logger log.Logger, reg *prometheus.Registry, tracer opentracing.Tracer, _ <-chan struct{}, _ bool) error {
-		return runCompact(g, logger, reg, component.Compact, *conf)
+		flagsMap := map[string]string{}
+
+		// Exclude kingpin default flags to expose only Thanos ones.
+		boilerplateFlags := kingpin.New("", "").Version("")
```

Comment:
sounds like some helper would be nice for this (:

Reply:
Yep, this was on my list, but I am not sure what would be the best place to write the helper.

Reply:
any (: something in this main package for a start would do (: